packages/kbn-ftr-common-functional-ui-services/services/toasts.ts

```diff
@@ -34,22 +38,55 @@ export class ToastsService extends FtrService {
     return { title, message };
   }
 
+  public async toastMessageByTestSubj(testSubj = 'csp:toast-success') {
+    const testSubjSvc = this.testSubjects;
+    return {
+      async getElement(): Promise<WebElementWrapper> {
+        return await testSubjSvc.find(testSubj);
+      },
+      async clickToastMessageLink(linkTestSubj = 'csp:toast-success-link') {
+        const element = await this.getElement();
+        const link = await element.findByTestSubject(linkTestSubj);
+        await link.click();
+      },
+    };
+  }
```

Comment:
I think this construction is not ideal. I know it has been taken over as-is from the old implementation, however it's always just called like this: `const toastMessage = await (await toasts.toastMessageByTestSubj()).getElement();` so there's no real benefit with the nested methods and IMO just makes the code more complicated to read, write and maintain. I'd suggest to refactor this bit, either in this PR or in a follow up.

Reply:
Agree with Robert. We should not expose WebDriver interfaces like `.getElement()`, but only toast specific  actions: close(), getText(), waitToDisappear().

Reply:
@pheyos @wayneseymour wdyt about removing `toast` from methods name?

`toasts.dismissAllToasts();` => `toasts.dismissAll();`
`await toasts.closeToast();` => `await toasts.close();`

Reply:
Sounds good to me :+1:

Reply:
I completely concur!

Reply:
Will be resolved by: https://github.com/elastic/kibana/issues/176570